src/cne/elector.rs

```diff
@@ -17,6 +18,236 @@ pub struct Elector {
 
 impl From<String> for Elector {
   fn from(html_string: String) -> Self {
-    todo!()
+    let document = Html::parse_fragment(&html_string);
+    let selector = Selector::parse("table tbody tr td table tbody tr:last-child td table tbody tr:nth-child(2) td table:first-child tbody tr").unwrap();
+    let field_selector = Selector::parse("tr td:last-child").unwrap();
+
+    let font_selector = Selector::parse("font").unwrap();
+    let bolt_selector = Selector::parse("b").unwrap();
+
+    let mut select = document.select(&selector);
+
+    let citizenship = select
+      .next()
+      .unwrap()
+      .select(&field_selector)
+      .next()
+      .unwrap()
+      .inner_html();
+
+    Elector {
+      citizenship: match citizenship.chars().next().unwrap() {
+        'E' => Citizenship::E,
+        _ => Citizenship::V,
+      },
+      identity: citizenship
+        .chars()
+        .skip(2)
+        .take(citizenship.len() - 2)
+        .collect(),
+      name: select
+        .next()
+        .unwrap()
+        .select(&field_selector)
+        .next()
+        .unwrap()
+        .select(&bolt_selector)
+        .next()
+        .unwrap()
+        .inner_html(),
+      state: select
+        .next()
+        .unwrap()
+        .select(&field_selector)
+        .next()
+        .unwrap()
+        .inner_html(),
+      town: select
+        .next()
+        .unwrap()
+        .select(&field_selector)
+        .next()
+        .unwrap()
+        .inner_html(),
+      parish: select
+        .next()
+        .unwrap()
+        .select(&field_selector)
+        .next()
+        .unwrap()
+        .inner_html(),
```

Comment:
Is there another way to accomplish this? I can see the same code many times, why not writing a function to make it more readable and more "DRY"?

Reply:
I already refactored the code dividing the selection into 3 functions